doc/api/child_process.markdown

```diff
@@ -392,76 +392,202 @@ controlling terminal.
-   have an underlying descriptor (file streams do not until the `'open'`
-   event has occurred).
-5. Positive integer - The integer value is interpreted as a file descriptor
-   that is is currently open in the parent process. It is shared with the child
-   process, similar to how `Stream` objects can be shared.
-6. `null`, `undefined` - Use default value. For stdio fds 0, 1 and 2 (in other
-   words, stdin, stdout, and stderr) a pipe is created. For fd 3 and up, the
-   default is `'ignore'`.
+The `options.stdio` property is used to configure the [standard streams][] of a
+child process. This allows a child process to receive input from devices like
+the keyboard and send output to devices like the terminal. Configuring
+`options.stdio` also allows the parent to connect a child to additional
+endpoints like files and sockets.
 
-Example:
+The `options.stdio` property accepts an `Array` where each index corresponds to
+the child's standard stream file descriptor. Indices 0, 1, and 2 correspond to
+the stdin, stdout, and stderr streams respectively. Indices 3 and up can be used
+to configure non-standard streams.
+
+The following are the possible values for the `options.stdio` configuration
+`Array`.
+
+* `Integer` file descriptor
+* `Stream` object
+* `'ignore'`
+* `'ipc'`
+* `'pipe'`
+* `null`
+* `undefined`
+
+For convenience, the `options.stdio` property also accepts one of the following
+`String` shorthands.
+
+| Shorthand   | Equivalent                       |
+|-------------|----------------------------------|
+| `'ignore'`  | `['ignore', 'ignore', 'ignore']` |
+| `'inherit'` | `[0, 1, 2]`                      |
+| `'pipe'`    | `['pipe', 'pipe', 'pipe']`       |
+
+When no value is provided, `options.stdio` defaults to `'pipe'`.
+
+##### `Integer` file descriptor
+
+Shares the parent process's open `Integer` file descriptor with a child process.
+The connection to the parent's file descriptor resource is duplicated and
+attached to a new file descriptor in the child. The new file descriptor
+corresponds to the index in the `options.stdio` configuration `Array`.
+
+The following example spawns a child process that shares the parent's stdin,
+stdout, and stderr `Integer` file descriptors.
 
     const spawn = require('child_process').spawn;
+    const child = spawn('node', ['--interactive'], { stdio: [0, 1, 2] });
+
+    child.on('close', function(code) {
+      process.exit(code);
+    });
+
+##### `Stream` object
+
+Shares a parent process's `Stream` object with a child process. These `Stream`
+objects must contain an open file descriptor. Therefore, valid objects are
+instances of the following classes.
+
+* `fs.ReadStream`
+* `fs.WriteStream`
+* `net.Socket`
+* `tty.ReadStream`
+* `tty.WriteStream`
+
+**NOTE:** Until the `'open'` event is emitted, `fs.ReadStream` and
+`fs.WriteStream` objects don't have an open file descriptor.
+
+The connection to the parent's `Stream` resource is duplicated and attached to a
```

Comment:
"The file descriptor in the parent's Stream resource is duplicated as a new file descriptor in the child."